src/ui/public/field_editor/field_editor.html

```diff
@@ -26,6 +26,15 @@
 
   <div ng-if="editor.field.scripted" class="form-group">
     <label>Language</label>
+    <div class="hintbox" ng-if="!editor.isSupportedLang(editor.field.lang)">
+      <h4>
+        <i class="fa fa-warning text-warning"></i> Deprecation Warning
+      </h4>
+      <p>
+        <span class="text-capitalize">{{editor.field.lang}}</span> is deprecated and support will be removed in the
+        next major version of Kibana and Elasticsearch. We recommend using Painless for new scripted fields.
```

Comment:
Linking to the Painless docs in the index-level warning is a great idea. Let's do the same here.